tests/std/tests/P0896R4_views_take_while/test.cpp

```diff
@@ -0,0 +1,493 @@
+    assert(ranges::equal(r, expected));
+
+    { // Validate take_while_view::pred
+        [[maybe_unused]] same_as<Pred> auto pred_copy = as_const(r).pred();
+        STATIC_ASSERT(noexcept(as_const(r).pred()));
+    }
+
+    const bool is_empty = ranges::empty(expected);
+
+    // Validate view_interface::empty and operator bool
+    STATIC_ASSERT(CanMemberEmpty<R> == forward_range<Rng>);
+    STATIC_ASSERT(CanBool<R> == CanEmpty<R>);
+    if constexpr (CanMemberEmpty<R>) {
+        assert(r.empty() == is_empty);
+        assert(static_cast<bool>(r) == !is_empty);
+    } else {
+        STATIC_ASSERT(CanEmpty<R> == CanSize<R>);
+        if constexpr (CanEmpty<R>) {
+            assert(ranges::empty(r) == is_empty);
+            assert(static_cast<bool>(r) == !is_empty);
+        }
+    }
+
+    STATIC_ASSERT(CanMemberEmpty<const R> == forward_range<const Rng>);
+    STATIC_ASSERT(CanBool<const R> == CanEmpty<const R>);
+    if constexpr (CanMemberEmpty<const R>) {
+        assert(as_const(r).empty() == is_empty);
+        assert(static_cast<bool>(as_const(r)) == !is_empty);
+    } else {
+        STATIC_ASSERT(CanEmpty<const R> == CanSize<const R>);
+        if constexpr (CanEmpty<const R>) {
+            assert(ranges::empty(as_const(r)) == is_empty);
+            assert(static_cast<bool>(as_const(r)) == !is_empty);
+        }
+    }
+
+    // Validate take_while_view::begin
+    STATIC_ASSERT(CanMemberBegin<R>);
+    STATIC_ASSERT(CanMemberBegin<const R>);
+    if (forward_range<V>) { // intentionally not if constexpr
+        const same_as<iterator_t<R>> auto i = r.begin();
+        if (!is_empty) {
+            assert(*i == *begin(expected));
+            assert(*r.begin() == *begin(expected));
+            assert(*r.begin() == *begin(expected));
+        }
+
+        if constexpr (copyable<V>) {
+            auto r2                              = r;
+            const same_as<iterator_t<R>> auto i2 = r2.begin();
+            if (!is_empty) {
+                assert(*i2 == *i);
+                assert(*r2.begin() == *i2);
+                assert(*r2.begin() == *i2);
+            }
+        }
+
+        const same_as<iterator_t<const R>> auto ic = as_const(r).begin();
+        if (!is_empty) {
+            assert(*ic == *begin(expected));
+            assert(*as_const(r).begin() == *begin(expected));
+            assert(*as_const(r).begin() == *begin(expected));
+        }
+
+        if constexpr (copyable<V>) {
+            auto rc2                                    = as_const(r);
+            const same_as<iterator_t<const R>> auto ic2 = rc2.begin();
+            if (!is_empty) {
+                assert(*ic2 == *ic);
+                assert(*rc2.begin() == *ic2);
+                assert(*rc2.begin() == *ic2);
+            }
+        }
+    }
+
+    // Validate take_while_view::end
+    STATIC_ASSERT(CanMemberEnd<R>);
+    STATIC_ASSERT(CanMemberEnd<const R>);

```

Comment:
Should be `== (range<const R> && LWG3450_thing_I_dont_remember)`

Reply:
It is obviously `indirect_unary_predicate<const Pred, iterator_t<const V>>`